aptos-move/aptos-vm/src/delta_ext/transaction.rs

```diff
@@ -65,6 +67,52 @@ impl TransactionOutputExt {
     pub fn into(self) -> (DeltaChangeSet, TransactionOutput) {
         (self.delta_change_set, self.output)
     }
+
+    /// Similar to `into()` but tries to apply delta changes as well.
+    /// TODO: ideally, we may want to expose this function to VM instead. Since
+    /// we do not care about rerunning the epilogue - it sufficies to have it
+    /// here for now.
+    pub fn into_transaction_output_with_status(
```

Comment:
merge the logic to either apply deltas or not here (based on status). If it becomes too long or ugly, can move delta application logic to a private apply_deltas method.